contracts/cw721-piggy-bank/src/contract.rs

```diff
@@ -32,11 +34,14 @@ pub fn instantiate(
 ) -> StdResult<Response> {
     cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
-    // TODO Validate denoms are formated correctly
-
+    // Validate denoms are formatted correctly
+    let unchecked_denom = UncheckedDenom::Native(msg.deposit_denom.clone());
+    let _checked_denom = unchecked_denom.into_checked(deps.as_ref()).map_err(|_| StdError::generic_err("Invalid deposit denom"))?;
+    
     // Save config info
-    DEPOSIT_DENOM.save(deps.storage, &msg.deposit_denom)?;
-    // TODO validate base_url is a real url
+    DEPOSIT_DENOM.save(deps.storage, &_checked_denom.to_string())?;
```

Comment:
See previous message.

```suggestion
    DEPOSIT_DENOM.save(deps.storage, &checked_denom.to_string())?;
```